site/docs/tutorial/sql_to_substrait.md

```diff
@@ -0,0 +1,245 @@
+SQL expression:
+
+```sql
+a > 10
+```
+
+An expression can be one of several things. `a` itself will be translated to
+a field reference:
+
+```json
+{
+  "selection": {
+    "directReference": {
+      "structField": {
+        "field": 1
+      }
+    },
+    "rootReference": {}
+  }
+}
+```
+
+!!! note
+    Protobuf may not serialize fields with integer type and value 0, since 0 is the default.
+    So if you instead saw `"structField": {}`, know that is is equivalent to 
+    `"structField": { "field": 0 }`.
+
+`10` will be translated to a literal expression:
+
+```json
+{
+  "literal": {
+    "i8": 10,
+    "nullable": true
+  }
+}
+```
+
+Then full expression `a > 10` will be a scalar function expression, and will 
+contain the expressions for `a` and `10` in it's `arguments` field:
+
+```json
+{
+  "scalarFunction": {
+    "functionReference": 2,
+    "outputType": {
+      "bool": {
+        "nullability": "NULLABILITY_NULLABLE"
+      }
+    },
+    "arguments": [
+      {
+        "value": {
+          "selection": {
+            "directReference": {
+              "structField": {
+                "field": 1
+              }
+            },
+            "rootReference": {}
+          }
+        }
+      },
+      {
+        "value": {
+          "literal": {
+            "i8": 10,
+            "nullable": true
+          }
+        }
+      }
+    ]
+  }
+}
+```
+
+The name of the function is defined by the `functionReference`, which points to
+the index of the extension function to use. This is serialized separately, which
+you will see later in serialized plans.
+
+```json
+{
+  "extensionFunction": {
+    "extensionUriReference": 1,
+    "functionAnchor": 2,
+    "name": "gt"
+  }
+}
+```
+
+## Relations
+
+An logical or physical plan is typically represented as a tree of nodes, such
+as filter, project, scan, or join. Each of these nodes is a **Relation** in 
+Substrait.
+
+For example, to read a named table, we use a `read` relation. This looks like:
+
+```json
+{
+  "read": {
+    "baseSchema": {
+      "names": [
+        "b",
+        "a"
+      ],
+      "struct": {
+        "types": [
+          {
+            "string": {
+              "nullability": "NULLABILITY_NULLABLE"
+            }
+          },
+          {
+            "i64": {
+              "nullability": "NULLABILITY_NULLABLE"
+            }
+          }
+        ],
+        "nullability": "NULLABILITY_REQUIRED"
+      }
+    },
+    "namedTable": {
+      "names": [
+        "my_database",
+        "my_table"
+      ]
+    }
+  }
+}
+```
+
+Read relations provide a schema in `baseSchema` as discussed in the prior section.
+The type of read is a named table, so the `namedTable` field is present with 
+`names` containing the list of name segments (`my_database.my_table`).
+
+Read relations are leaf nodes, so other relations will have them nested within.
+For example, a filter relation of (`a > 10`) will look like:
```

Comment:
Maybe...

Read relations are leaf nodes.  Leaf nodes don't depend on any other node for data and usually represent a source of data in our plan.  Leaf nodes are then typically used as input for other nodes that manipulate the data.

Reply:
Do you think we need to refer to the docs https://substrait.io/relations/logical_relations/ where has a table shows the input and output number of each relation? Like read relation here require 0 input so that's why it's a leaf relation.